### ISSUE TITLE
merge: share a shell?

### ISSUE THREAD

**Issue by maowtm** [modus-continens/modus#165]
I just realized that it might actually be more helpful to define merge as making all the run commands share a single shell execution. This makes writing things like the following very easy:

```Modusfile
install_foo(...) :- (
  run("savedAptMark=$(apt-mark showmanual)"),
  build_foo(...),
  run("apt-mark manual $savedAptMark")
)::merge.
```

Although this does means that if we want to move away from the "shell-escape copy() paths and turn the whole thing into a shell script" approach things will get very complicated.

**Comment by mechtaev** [comment]
This would mean that using and not using `::merge` will result in different builds.

**Comment by micromaomao** [comment]
Yeah that's a good point… Currently to do this I'm just saving the variable to a file, which I guess works well enough. But I wonder if we want to have an operator specifically to do this kind of single-shell merges, just to simplify situations like these.

**Comment by mechtaev** [comment]
Is it going to be shallow, or deep, i.e. affecting `build_foo` as well? If deep, it will break code modularity.
